Currency Input component

juligobbi-fluxit opened the pull request:
It is created the Currency Input component. It is pending to take the functionality of the input

BrandonDenham commented on src/components/currency/Currency.tsx:
```diff
@@ -0,0 +1,81 @@
+    onChange,
+    detail,
+}) => {
+    const currency = 'USD';
```
we need a currencyPosition prop that defaults to left since some currencies show up on the right of an input

BrandonDenham replied:
we also need a decimalPlaces prop that defaults to 0.  currencies will need to have decimals in some cases.

juligobbi-fluxit replied:
Were solved the three of them

BrandonDenham commented on src/components/currency/Currency.tsx:
```diff
@@ -0,0 +1,81 @@
+                    type="text"
+                    value={value}
+                    css={input(theme, error)}
+                    onChange={e => handleChange(e, parseInt(e.target.value))}
```
you shouldnt have any logic inline of the onChange prop, should move any parsing to the handleChange function itself

Also, you cant use "parseInt" since we need to support decimals

something like:

Number(Number.parseFloat(value).toFixed(props.decimalPlaces));
or something

juligobbi-fluxit replied:
I took the logic inline of the onChange prop out. The decimal is pending.

BrandonDenham commented on src/components/currency/Currency.tsx:
```diff
@@ -0,0 +1,81 @@
+            {error && typeof error === 'string' && (
+                <span css={errorSpan(theme)}>{error}</span>
+            )}
+            {error && Array.isArray(error) && (
```
what if array is empty

BrandonDenham replied:
basically should calculate the errortext outside of the return then inside the return can be simple:

{errorText && <span css={errorSpan(theme)}>{errorText}</span>}

juligobbi-fluxit replied:
Was solved

juligobbi-fluxit commented:
**Regarding requested changes, in the following commit:**

**is Solved:**
- Was created a "CurrencyProps" interface that extends "FieldProps"
- Added currencyType as a prop
- Added currencyPosition as a prop
- Added decimalPlaces as a prop that defaults to 0.
- Take outside the logic inline of the onChange prop
- Error function:
Take outside the error function
Extended the function (empty, single, multiple)

**Also added 2 variants more:**
- Added size as a prop and with an example in the stories. To use the extra small size in tables.
- Added position as a prop and with an example in the stories. To use the position left or right for the type of currency.


**is Pending:**
- Need to support decimals. Example:
	Number(Number.parseFloat(value).toFixed(props.decimalPlaces));

juligobbi-fluxit commented on src/components/currency/Currency.tsx:
```diff
@@ -0,0 +1,81 @@
+ * @param  props.disabled - For usage as disabled.
+ */
+
+const Currency: React.FC<FieldProps<number>> = ({
```
Was solved

BrandonDenham reviewed (approved):
nice job!
